src/dev.rs

```diff
@@ -250,7 +242,9 @@ pub struct MockProver<F: Group + Field> {
     // The instance cells in the circuit, arranged as [column][row].
     instance: Vec<Vec<F>>,
 
-    permutations: Vec<permutation::keygen::Assembly>,
+    permutation: permutation::keygen::Assembly,
+
+    upper_bound_cell_index: usize,
```

Comment:
Document what this is.

Reply:
Adding documentation for it. Note that it's documented elsewhere in `plonk::keygen`, this is just the developer tooling, but it can't hurt to have additional docs.

Reply:
I'm not sure why it's called `something_cell_index`, though. Also "upper bound" is ambiguous as to whether it's the last usable row or the last plus one (whereas `usable_rows` is unambiguous, at least if you think about it for a bit).

Reply:
Okay, I've [changed it](https://github.com/zcash/halo2/pull/316/commits/bd29ebdb057768234104846453b4545cb3e41a60) to `usable_rows` and made it a `RangeTo<usize>` (per @str4d's suggestion).